### PR TITLE
Adding pipeline to deploy the new image on master

### PR DESCRIPTION
**Pull request by moizjv**
@ryanckoch Please review.

- Pipeline builds and pushes to quay
- Deploys to k8s sjc prod

FYI @saucelabs/autdev

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  stage('Deploy if prod') {
+    container('helm') {
+      if (env.BRANCH_NAME == 'master') {
+          withCredentials([file(credentialsId: 'SJC_PROD_DEFAULT_CONFIG', variable: 'CONFIG_FILE')]) {
+            sh """#!/bin/bash -ex
+                  mkdir ~/.kube
+                  mv ${CONFIG_FILE} ~/.kube/config
+            """
+          }
+          sh 'helm upgrade -i helm-charts k8s/chart/ -f k8s/conf/prod.yaml --namespace=private --set image.tag=${env.SHORT_COMMIT}'
```
Comment (ryanckoch) [inline Jenkinsfile]: I would just set
```
 ingress:
    enabled: true
```
in the k8s/chart/values.yaml and get rid of the k8s/conf directory completely

**Comment by ryanckoch** [comment]
👍

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      sh "docker build -t quay.io/saucelabs/helm-charts:${env.SHORT_COMMIT} ."
+      withCredentials([usernamePassword(credentialsId:'saucelabs-quay-jenkins-bot',
+        usernameVariable: 'Q_USERNAME', passwordVariable: 'Q_PASSWORD')]) {
+          sh "docker login quay.io --username $Q_USERNAME --password $Q_PASSWORD"
+          sh "docker push quay.io/saucelabs/helm-charts:${env.SHORT_COMMIT}"
+      }
+    }
+  }
+
+  stage('Deploy if prod') {
```
Comment (bernii) [inline Jenkinsfile]: shouldn't that be 'deploy if master branch' ?

Comment (moizjv) [inline Jenkinsfile]: Good catch.

**Pull request merged**